src/proposal.mligo

```diff
@@ -260,32 +170,14 @@ let vote(votes, config, store : vote_param_permited list * config * storage): re
     let store = ensure_proposal_voting_period (proposal, config.voting_period, store) in
     let store = submit_vote (proposal, vote_param, author, config.voting_period, store) in
     store
-    in
-  ( ([] : operation list)
-  , List.fold accept_vote votes store
-  )
+  in
+  (nil_op, List.fold accept_vote votes store)
 
-// -----------------------------------------------------------------
-// Admin entrypoints
-// -----------------------------------------------------------------
 
 [@inline]
 let burn_frozen_token (tokens, addr, store : nat * address * storage): storage =
   let (ledger, total_supply) = debit_from(tokens, addr, store.frozen_token_id, store.ledger, store.total_supply)
-  in {store with ledger = ledger; total_supply = total_supply}
-
-// Burn up to desired_burn_amount of tokens. The desired burn amount comprises
-// slash amount calculated by "config.rejected_proposal_return_value" and
-// the fixed fee payed for the proposal. The case when the desired burn amount
-// is larger than the available frozen tokens is possible because the contract
-// administrator can transfer the proposer's frozen tokens.
-[@inline]
-let burn_what_possible (desired_burn_amount, frozen_tokens, addr, store : nat * nat * address * storage): storage =
-  let to_burn =
-    if frozen_tokens >= desired_burn_amount
-    then desired_burn_amount
-    else frozen_tokens
-  in burn_frozen_token (to_burn, addr, store)
```

Comment:
> the user might have already unfreezed some of their tokens

Yes, but not the ones he stakes to create the proposal, `unfreeze` will prevent that.

So since this needs only to be able to burn tokens that were previously staked to propose this should never result in an issue.

I added a `"cannot unfreeze staked tokens"` test for that now.